apis/projectcontour/v1/httpproxy.go

```diff
@@ -18,6 +18,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// HttpVersion is an alias to enforce validation
+// +kubebuilder:validation:Enum=h2;http/1.1
```

Comment:
Since we already have these constants defined in the ContourConfiguration struct (in v1alpha1) and the Contour config file, let's use the same values for them here for consistency:
```suggestion
// +kubebuilder:validation:Enum=HTTP/2;HTTP/1.1
```

Yes, this will then require a translation to the Envoy-accepted values, but we already have code for this in `cmd/contour/parseDefaultHTTPVersions` and `internal/envoy/v3/ProtoNamesForVersions` that we should be able to reuse, possibly with some refactoring required.

Reply:
The `ProtoNamesForVersions` function converts from `int32` codecs to `["http/1.1", "h2"]` .

The `parseDefaultHTTPVersions` converts from `["HTTP/1.1", "HTTP/2"]`  to `int32` codecs.

Note that the desired format for Envoy configuration is ["http/1.1", "h2"]. So, why use the ["HTTP/1.1", "HTTP/2"] format at all instead of directly using ["http/1.1", "h2"]? 😅 

Additionally, those functions cannot be easily imported since they're private, or importing them leads to import cycles unless we move them into a util-like package.

Reply:
This [comment](https://github.com/projectcontour/contour/pull/5802#discussion_r1442300257) already answers my question. So thanks. I'll refactor.